builder/detection_order.go

```diff
@@ -0,0 +1,18 @@
+package builder
+
+import (
+	"github.com/buildpacks/pack/internal/dist"
+)
+
+type DetectionOrderEntry struct {
+	dist.BuildpackRef
```

Comment:
```suggestion
	dist.BuildpackRef `yaml:",inline"`
```

^ this fixes the output so that there is no empty `buildpackref: {}`